File: ProjectP/src/split.py
#coding:utf-8

FILE="../data/core_SUW.txt"

"""
BCCWJ_coreから、行ごとの分かち書きを出力する
作成日：2016/5/31
作成者：久保大輝

"""
FILE="../data/core_SUW.txt"

#registerHash={'OC':'Yahoo!知恵袋'}


def main():
	f=open(FILE,encoding='utf-8')
	for line in f:
#		print(line.split('\t'))
		regiID=line.split('\t')[0]
		sampleID=line.split('\t')[1]
		beginLabel=line.split('\t')[9]
		lemma=line.split('\t')[12]
		subLemma=line.split('\t')[14]
		wType=line.split('\t')[15]
		pos=line.split('\t')[16]
		orthBase=line.split('\t')[21]
		originalText=line.split('\t')[22]

#		print(lemma+","+wType)
		for char in lemma:
			wType=checkType(char)	#1文字ずつに分割して文字種判別
	f.close()

if __name__=='__main__':
	main()

File: ProjectP/src/split.jl
#-*- coding:utf-8 -*-

using DataFrames

# coreデータを学習用とテスト用に分割
# 第二引数はclass-a1のIDを列挙したファイル
function split_core_data(source, idf)
  core = open(source, "r")
#  class_a1 = readtable(idf, separator = '\t', header = false)[1]
	fp=open(idf)
	
#	class_a1:IDの配列
  train, test = open("../data/train_original.txt", "w"), open("../data/test_original.txt", "w")
  println("分割中(/・ω・)/・・・")
  for line in eachline(core)
    data = split(line, "\t")
    if data[2] in class_a1
      write(test, line)
    else
      write(train, line)
    end
  end
  close(train)
  close(test)
end

# WordSegmentation用の形式に変換
# めんどくさいので型指定してない
function make_wakati(ifile,ofile)
  f,o = open(ifile, "r"), open(ofile, "w")
  println("行ごとの分かち書きに変換中(/・ω・)/・・・")
  sentence = []
  for line in eachline(f)
    data = split(line, "\t")
    word, head = data[24],data[10]
    if (sentence != [] && head == "B")
      write(o, join(sentence, " ")*"\n")
      sentence = [word]
    else
      push!(sentence, word)
    end
  end
end

split_core_data("../data/core_SUW.txt", "../data/ClassA-1.txt")　
make_wakati("../data/test_original.txt", "../data/test_wakati.txt")
make_wakati("../data/train_original.txt", "../data/train_wakati.txt")
